src/snowflake/snowpark/_internal/ast.py

```diff
@@ -121,6 +121,9 @@ def decode_ast_response_from_snowpark(res: dict, session_parameters: Any) -> Any
 
 
 class AstBatch:
+    # Function used to generate request IDs. This is overridden in some tests.
+    request_id_gen_fn = uuid.uuid4
```

Comment:
Trivial topic: I know fields are usually nouns, but in true OOP (like in Python), there's no substantial difference between fields and methods anyway. Should this name simply be `generate_request_id`?